src/main/java/gregtech/api/block/machines/BlockMachine.java

```diff
@@ -262,34 +262,37 @@ public void onBlockPlacedBy(World worldIn, @NotNull BlockPos pos, @NotNull IBloc
 
         MetaTileEntity sampleMetaTileEntity = registry.getObjectById(stack.getItemDamage());
-        if (holder != null && sampleMetaTileEntity != null) {
-            // TODO Fix this
-            if (stack.hasDisplayName() && holder instanceof MetaTileEntityHolder) {
-                ((MetaTileEntityHolder) holder).setCustomName(stack.getDisplayName());
-            }
-            MetaTileEntity metaTileEntity = holder.setMetaTileEntity(sampleMetaTileEntity);
-            var stackTag = stack.getTagCompound();
-            if (stackTag != null && !stackTag.isEmpty()) {
-                if (stackTag.hasKey(GregtechDataCodes.BLOCK_ENTITY_TAG)) {
-                    var blockTag = stackTag.getCompoundTag(GregtechDataCodes.BLOCK_ENTITY_TAG);
-                    String customName = blockTag.getString(GregtechDataCodes.CUSTOM_NAME);
-                    if (!customName.isEmpty())
-                        ((MetaTileEntityHolder) holder).setCustomName(customName);
-
-                    var mteTag = blockTag.getCompoundTag(GregtechDataCodes.TAG_KEY_MTE);
-                    List<String> removed = new ArrayList<>();
-                    for (var key : mteTag.getKeySet()) {
-                        var trait = metaTileEntity.getMTETrait(key);
-                        if (trait == null) continue;
-
-                        removed.add(key);
-                    }
-                    removed.forEach(mteTag::removeTag);
-                    metaTileEntity.readFromNBT(mteTag);
-                } else {
-                    metaTileEntity.initFromItemStackData(stackTag);
+        if (holder == null || sampleMetaTileEntity == null)
+            return;
+
+        // TODO Fix this
+        if (stack.hasDisplayName() && holder instanceof MetaTileEntityHolder) {
+            ((MetaTileEntityHolder) holder).setCustomName(stack.getDisplayName());
+        }
+        var stackTag = stack.getTagCompound();
+        NBTTagCompound mteTag = null;
+        if (stackTag != null && !stackTag.isEmpty()) {
+            if (stackTag.hasKey(GregtechDataCodes.BLOCK_ENTITY_TAG)) {
+                var blockTag = stackTag.getCompoundTag(GregtechDataCodes.BLOCK_ENTITY_TAG);
+                String customName = blockTag.getString(GregtechDataCodes.CUSTOM_NAME);
+                if (!customName.isEmpty())
+                    ((MetaTileEntityHolder) holder).setCustomName(customName);
+
+                mteTag = blockTag.getCompoundTag(GregtechDataCodes.TAG_KEY_MTE);
+                List<String> removed = new ArrayList<>();
+                for (var key : mteTag.getKeySet()) {
+                    var trait = sampleMetaTileEntity.getMTETrait(key);
+                    if (trait == null) continue;
+
+                    removed.add(key);
                 }
+                removed.forEach(mteTag::removeTag);
+            } else {
+                sampleMetaTileEntity.initFromItemStackData(stackTag);
```

Comment:
Shouldn't this be called on the actual MTE set/created by the holder, instead of the sample one?

Reply:
yeah, calling it on the sample would mean the loaded date gets lost. it's fixed now